src/components/AddEditForm/index.tsx

```diff
@@ -20,41 +20,45 @@ const yearOptions = DropdownFilterConstant.year.options
     return <option value={option}>{option}</option>;
   });
 
-function AddEditForm(props: AddEditFormProps) {
+const AddEditForm = (props: AddEditFormProps) => {
   const isDisabled = props.type === "EDIT";
 
-  const [filterState, setFilterState] = useState({
+  const [courseState, setCourseState] = useState({
     name: props.course.name,
     code: props.course.code,
     credits: props.course.credits,
     branch: props.course.branch,
     year: props.course.year,
   });
 
-  const submitHandler = useCallback((event: any) => {
-    event.preventDefault();
-    const curCourse = {
-      name: filterState.name,
-      prof: "John Doe",
-      strength: props.course.strength,
-      branch: filterState.branch,
-      year: filterState.year,
-      code: filterState.code,
-      credits: filterState.credits,
-    };
-    props.onSubmit(curCourse);
-  }, []);
-
-  const handleOnChange = useCallback((event: any) => {
-    let value = event.target.value;
-    if (event.target.name === "year" || event.target.name === "credits") {
-      value = parseInt(event.target.value);
-    }
+  const submitHandler = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      props.onSubmit({
```

Comment:
SHould check if `props.onSubmit` is undefined or not, always make it a habit of checking functions from prop